pkg/profiling/task/oncpu/runner.go

```diff
@@ -0,0 +1,311 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package oncpu
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/perf"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/apache/skywalking-rover/pkg/logger"
+	"github.com/apache/skywalking-rover/pkg/process/api"
+	"github.com/apache/skywalking-rover/pkg/profiling/task/base"
+	"github.com/apache/skywalking-rover/pkg/tools"
+	"github.com/apache/skywalking-rover/pkg/tools/profiling"
+
+	"golang.org/x/sys/unix"
+
+	v3 "skywalking.apache.org/repo/goapi/collect/ebpf/profiling/v3"
+)
+
+// $BPF_CLANG and $BPF_CFLAGS are set by the Makefile.
+// nolint
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc $BPF_CLANG -cflags $BPF_CFLAGS bpf $REPO_ROOT/bpf/profiling/oncpu.c -- -I$REPO_ROOT/bpf/include
```

Comment:
You set `REPO_ROOT` with an absolute path in the makefile as `export REPO_ROOT := /skywalking-rover`, this works in `make container-generate` because you mount the repo to `/skywalking-rover` in the container but it mostly will fail in others' machines (like mine), I have to put the repo in `/skywalking-rover` which is unrealistic. Why not consider using relative path? Or, if we cannot run all the commands in non-Linux machines, I'd rather not to have the `generate` command at all, just use `container-generate` and run all commands in containers.

Reply:
Yes, I'm working on fixing it. I'm still searching which GitHub action supports the LLVM and Clang for building this project.

Reply:
Should update the `REPO_ROOT` value to `$(REPODIR)`.

Reply:
> Yes, I'm working on fixing it. I'm still searching which GitHub action supports the LLVM and Clang for building this project.

Don't you plan to build the project in Docker container? I'm afraid for such kind of projects, without Docker container with pre-set environment it's hard for people to build

Reply:
I have added the `container-all` command, including `lint`, `test`, and `build` in a Docker container. could I just use it with CI? Because in the CI we are already in a ubuntu environment, so I think they don't need to using a docker container.

Reply:
> I have added the `container-all` command, including `lint`, `test`, and `build` in a Docker container. could I just use it with CI? Because in the CI we are already in a ubuntu environment, so I think they don't need to using a docker container.

I prefer we unified the commands in both CI and local machine. Also, to set up all necessary things for this project in GitHub CI environment is not an easy thing in my opinion and not necessary.

Do you mean we can run `make all` in Linux system without container too?

Reply:
`make container-all` means set up a docker container that contains all the software need to be used to build and Linux based.  
`make all` means the user has all software and build on Linux. 
Actually, the `make container-all` is setup in a docker and running the `make all` command.